### PR TITLE
timeAndCount -> count & reduce use

### PR DESCRIPTION
**Pull request by kenoir**
### What is this PR trying to achieve?

Reduce the use of CloudWatch metrics where they are not useful and achieve cost saving from this.

### Who is this change for?

🤑 

### Have the following been considered/are they needed?

- [ ] Deployed new versions
- [ ] Run `terraform apply`.

### CODE REVIEW

File: catalogue_pipeline/id_minter/src/main/scala/uk/ac/wellcome/platform/idminter/steps/IdEmbedder.scala [hunk 1]
```diff
@@ -18,14 +18,8 @@ class IdEmbedder @Inject()(
...
-  def embedId(json: Json): Future[Json] = {
-    metricsSender.timeAndCount(
-      "generate-id",
-      () =>
-        Future {
-          iterate(root.each, addIdentifierToJson(json))
-      }
-    )
+  def embedId(json: Json): Future[Json] = Future {
+    iterate(root.each, addIdentifierToJson(json))
```
Comment (alexwlchan) [inline catalogue_pipeline/id_minter/src/main/scala/uk/ac/wellcome/platform/idminter/steps/IdEmbedder.scala]: You can remove metricsSender from IdEmbedder entirely now.

File: catalogue_pipeline/ingestor/src/main/scala/uk/ac/wellcome/platform/ingestor/services/WorkIndexer.scala [hunk 2]
```diff
@@ -30,44 +30,35 @@ class WorkIndexer @Inject()(
...
-  def indexWork(work: IdentifiedWork,
-                esIndex: String,
-                esType: String): Future[Any] = {
+  def indexWork(work: IdentifiedWork, esIndex: String, esType: String) = {
 
-    metricsSender.timeAndCount[Any](
-      "ingestor-index-work",
-      () => {
-        info(s"Indexing work ${work.canonicalId}")
+    info(s"Indexing work ${work.canonicalId}")
```
Comment (alexwlchan) [inline catalogue_pipeline/ingestor/src/main/scala/uk/ac/wellcome/platform/ingestor/services/WorkIndexer.scala]: And WorksIndexer. (And probably others, I won’t keep checking.)

File: sbt_common/messaging/src/main/scala/uk/ac/wellcome/messaging/sqs/SQSStream.scala [hunk 3]
```diff
@@ -67,10 +67,6 @@ class SQSStream[T] @Inject()(actorSystem: ActorSystem,
         logger.error(
           s"Unrecognised failure while processing message ${message.getMessageId}",
           exception)
-        metricsSender.incrementCount(
-          metricName = s"${streamName}_MessageProcessingFailure",
-          count = 1.0
-        )
```
Comment (alexwlchan) [inline sbt_common/messaging/src/main/scala/uk/ac/wellcome/messaging/sqs/SQSStream.scala]: Okay, maybe we should keep this, oops.

Comment (alicefuzier) [reply sbt_common/messaging/src/main/scala/uk/ac/wellcome/messaging/sqs/SQSStream.scala]: If this goes away, We can get rid of GracefulFailureException cause this is the only difference in behaviour between the two types of exception

Comment (alexwlchan) [reply sbt_common/messaging/src/main/scala/uk/ac/wellcome/messaging/sqs/SQSStream.scala]: Ah no, it's covered above.

Comment (alexwlchan) [reply sbt_common/messaging/src/main/scala/uk/ac/wellcome/messaging/sqs/SQSStream.scala]: There’s also the difference in logging – a GracefulFailureException logs a WARN and a message, whereas an unrecognised exception includes the full stack trace for extra debugging.

**Comment by alexwlchan** [comment]
Part of #2177.

**Pull request merged**